system/lib/libc/musl/src/unistd/pread.c

```diff
@@ -4,7 +4,21 @@
 
 ssize_t pread(int fd, void *buf, size_t size, off_t ofs)
 {
+#if __EMSCRIPTEN__
+	__wasi_iovec_t iov = {
+		.buf = buf,
+		.buf_len = size
+	};
+	size_t num;
+	__wasi_errno_t error =
+			__wasi_fd_pread(fd, &iov, 1, ofs, &num);
+	if (error != __WASI_ERRNO_SUCCESS) {
+		return __wasi_syscall_ret(error);
```

Comment:
it's safe to call the helper `wasi_syscall_ret` all the time, which makes it simpler. See e.g. readv,
```c
	size_t num;
	if (__wasi_syscall_ret(__wasi_fd_read(fd, (struct __wasi_iovec_t*)iov, count, &num))) {
		num = -1;
	}
	return num;
```